validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/DepositProvider.java

```diff
@@ -134,7 +139,8 @@ public void onMinGenesisTimeBlock(MinGenesisTimeBlockEvent event) {}
     // the generated proofs are valid
     checkRequiredDepositsAvailable(eth1DepositCount, eth1DepositIndex);
 
-    UInt64 latestDepositIndexWithMaxBlock = eth1DepositIndex.plus(MAX_DEPOSITS);
+    UInt64 latestDepositIndexWithMaxBlock =
+        eth1DepositIndex.plus(specProvider.getMaxDeposits(state.getSlot()));
```

Comment:
nit: Could we/should we just pass the state to `getMaxDeposits`? Would solve the slot/epoch issue nicely.

Reply:
i've started moving things to use atState in the last push, i'll check if this function is used in many places...